e2e_tests/aws_ec2/test_ec2_deployment.py

```diff
@@ -0,0 +1,58 @@
+import logging
+import uuid
+import json
+
+from sklearn import datasets
+
+from bentoml.yatai.deployment.aws_utils import get_default_aws_region
+from e2e_tests.cli_operations import delete_deployment
+from e2e_tests.aws_ec2.utils import (
+    run_aws_ec2_create_command,
+    wait_for_healthy_targets,
+    send_test_data_to_multiple_endpoint,
+)
+
+logger = logging.getLogger('bentoml.test')
+
+
+def test_aws_ec2_deployment(iris_clf_service):
+    random_hash = uuid.uuid4().hex[:6]
+    deployment_name = f'tests-ec2-e2e-{random_hash}'
+    deployment_namespace = "test"
+    deployment_region = get_default_aws_region()
+
+    create_deployment_command = [
+        'bentoml',
+        'ec2',
+        'deploy',
+        deployment_name,
+        '-b',
+        iris_clf_service,
+        '--namespace',
+        deployment_namespace,
+        '--region',
+        deployment_region,
+        '--verbose',
+    ]
+
+    try:
+        deployment_endpoint = run_aws_ec2_create_command(create_deployment_command)
+        instance_adresses = wait_for_healthy_targets(
+            name=deployment_name,
+            namespace=deployment_namespace,
+            region=deployment_region,
+        )
+
+        assert deployment_endpoint, "AWS EC2 deployment creation should success"
+        assert instance_adresses, "AWS EC2 deployment should have all targets healthy"
+
+        iris = datasets.load_iris()
+        sample_data = iris.data[0:1]
+        results = send_test_data_to_multiple_endpoint(
```

Comment:
Can we also add a test case here which sends data to the ELB endpoint?

Reply:
It's already appended to list passed to that function, just above this line.

Reply:
got it!